lib/events/filesessions/fileasync_test.go

```diff
@@ -659,25 +661,36 @@ func emitStream(ctx context.Context, t *testing.T, streamer events.Streamer, inE
 }
 
 // readStream reads and decodes the audit stream from uploadID
+// Additionally, readStream will remove any duplicate audit events (by index) encountered.
 func readStream(ctx context.Context, t *testing.T, uploadID string, uploader *eventstest.MemoryUploader) []apievents.AuditEvent {
 	t.Helper()
 
 	parts, err := uploader.GetParts(uploadID)
 	require.NoError(t, err)
 
 	var outEvents []apievents.AuditEvent
-	var reader *events.ProtoReader
-	for i, part := range parts {
-		if i == 0 {
-			reader = events.NewProtoReader(bytes.NewReader(part))
-		} else {
-			err := reader.Reset(bytes.NewReader(part))
-			require.NoError(t, err)
-		}
+	for _, part := range parts {
+		reader := sessionrecording.NewReader(bytes.NewReader(part))
+
 		out, err := reader.ReadAll(ctx)
 		require.NoError(t, err, "part crash %#v", part)
 
+		require.NoError(t, reader.Close(), "error closing session recording reader")
+
 		outEvents = append(outEvents, out...)
 	}
-	return outEvents
+
+	// sort audit events by index
+	slices.SortStableFunc(outEvents, func(a apievents.AuditEvent, b apievents.AuditEvent) int {
```

Comment:
```suggestion
	slices.SortFunc(outEvents, func(a apievents.AuditEvent, b apievents.AuditEvent) int {
```

Reply:
Changed. Still feels wrong that I'm having to implement sort/uniq in a test function to drop events with duplicate indices to make a test case pass. More context in this discussion:  https://github.com/gravitational/teleport/pull/50501#discussion_r1901340534 Will investigate more tomorrow. I might be misunderstanding something.